src/Tizen.Wearable.CircularUI.Forms.Renderer/CircleSurfaceViewRenderer.cs

```diff
@@ -0,0 +1,128 @@
+                var box = new ObservableBox(XForms.NativeParent);
+                box.SetLayoutCallback(OnLayout);
+                _surfaceLayout = new ELayout(box);
+                _circleSurface = new CircleSurface(_surfaceLayout);
+                _circleSurfaceItems = new Dictionary<ICircleSurfaceItem, ICircleWidget>();
+                box.PackEnd(_surfaceLayout);
+                _surfaceLayout.Show();
+                SetNativeControl(box);
+            }
+
+            if (e.NewElement != null)
+            {
+                var items = e.NewElement.CircleSurfaceItems as ObservableCollection<ICircleSurfaceItem>;
+                items.CollectionChanged += OnCircleSurfaceItemsChanged;
+                foreach (var item in items)
+                {
+                    AddCircleSurfaceItem(item);
+                }
+            }
+
+            if (e.OldElement != null)
+            {
+                var items = e.OldElement.CircleSurfaceItems as ObservableCollection<ICircleSurfaceItem>;
+                items.CollectionChanged -= OnCircleSurfaceItemsChanged;
+            }
+
+            base.OnElementChanged(e);
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (Element != null)
+            {
+                var items = Element.CircleSurfaceItems as ObservableCollection<ICircleSurfaceItem>;
+                items.CollectionChanged -= OnCircleSurfaceItemsChanged;
+            }
+            base.Dispose(disposing);
+        }
+
+        void OnLayout()
+        {
+            var rect = Control.Geometry;
+            Element.Layout(rect.ToDP());
+            _surfaceLayout.Geometry = rect;
+        }
+
+        void OnCircleSurfaceItemsChanged(object sender, NotifyCollectionChangedEventArgs e)
+        {
+            if (e.Action == NotifyCollectionChangedAction.Add ||
+                e.Action == NotifyCollectionChangedAction.Replace)
+            {
+                foreach (ICircleSurfaceItem item in e.NewItems)
+                    AddCircleSurfaceItem(item);
+            }
+            if (e.Action == NotifyCollectionChangedAction.Remove ||
+                e.Action == NotifyCollectionChangedAction.Replace)
+            {
+                foreach (ICircleSurfaceItem item in e.OldItems)
+                    RemoveCircleSurfaceItem(item);
+            }
+        }
+
+        void AddCircleSurfaceItem(ICircleSurfaceItem item)
+        {
+            if (item is CircleProgressBarSurfaceItem)
+            {
+                var widget = new CircleProgressBarSurfaceItemImplements(item as CircleProgressBarSurfaceItem, _surfaceLayout, _circleSurface);
+                _circleSurfaceItems[item] = widget;
+            }
+            else if (item is CircleSliderSurfaceItem)
+            {
+                var widget = new CircleSliderSurfaceItemImplements(item as CircleSliderSurfaceItem, _surfaceLayout, _circleSurface);
+                _circleSurfaceItems[item] = widget;
+            }
```

Comment:
Why not support `CircleSurfaceItem` it also public class that implement ICirculeSurfaceItem

Reply:
I think `CircleSurfaceItem` was initially designed as an abstract base class. But it seems to be a public class by mistake. There is no corresponding native control. 🤔